tests/integration/llm/sagemaker-endpoint-tests.py

```diff
@@ -0,0 +1,176 @@
+import sagemaker
+import boto3
+from sagemaker.djl_inference import DJLModel, HuggingFaceAccelerateModel, DeepSpeedModel, FasterTransformerModel
+from sagemaker.huggingface import HuggingFaceModel
+from sagemaker.utils import unique_name_from_base
+from argparse import ArgumentParser
+
+parser = ArgumentParser(
+    description=
+    "This Script deploys a model with predefined configuration to a SageMaker Inference Endpoint"
+)
+parser.add_argument(
+    "model_name",
+    help="The predefined model and configuration to use for deployment")
+parser.add_argument("test_case",
+                    help="The test case to execute",
+                    choices=["djl", "no_code", "djl_mme"])
+
+SINGLE_MODEL_ENDPOINT_CONFIGS = {
+    "stable-diffusion-2-1-base": {
+        "model_id": "stabilityai/stable-diffusion-2-1",
+        "model_kwargs": {
+            "dtype": "fp16",
+            "number_of_partitions": 1,
+        },
+        "payload": {
+            "prompt": "A picture of a cowboy in space"
+        },
+        "deserializer": sagemaker.deserializers.BytesDeserializer()
+    },
+    "gpt2-xl": {
+        "model_id": "gpt2-xl",
+        "model_kwargs": {
+            "dtype": "fp32",
+            "number_of_partitions": 2,
+        }
+    },
+    "opt-1-3-b": {
+        "model_id": "s3://djl-llm/opt-1.3b/",
+        "model_kwargs": {
+            "dtype": "fp32",
+            "number_of_partitions": 1,
+        },
+        "cls_to_use": HuggingFaceAccelerateModel,
+    },
+    "flan-t5-xxl": {
+        "model_id": "s3://djl-llm/flan-t5-xxl/",
+        "model_kwargs": {
+            "dtype": "fp32",
+            "tensor_parallel_degree": 4,
+        },
+        "cls_to_use": FasterTransformerModel,
+    },
+    "gpt-j-6b": {
+        "model_id": "s3://djl-llm/gpt-j-6b/",
+        "model_kwargs": {
+            "dtype": "bf16",
+            "tensor_parallel_degree": 2,
+            "parallel_loading": True,
+        },
+        "cls_to_use": DeepSpeedModel,
+    },
+}
+
+HUGGING_FACE_NO_CODE_CONFIGS = {
+    "gpt-neo-2-7-b": {
+        "env": {
+            "HF_MODEL_ID": "EleutherAI/gpt-neo-2.7B",
+            "TENSOR_PARALLEL_DEGREE": "1",
+        },
+        "image_uri":
+        sagemaker.image_uris.retrieve(framework="djl-deepspeed",
```

Comment:
instead of testing public image, maybe we can point to our Alpha container repo to fetch the image there.

Reply:
So the test will cover the case when we did CVEs

Reply:
We can add a flag here to test on the nightly image, but I'm not sure if that should be the default. 

In my mind, this test suite can serve two purposes:
- Testing new changes in the Python SDK against our released DLCs
- Testing existing Python SDK against our nightly DLCs

I'll modify the workflow to include both options, but let's take that as a follow up. I'm not able to verify the workflow dispatch functionality here until the workflow exists in master branch.